python/nano/src/bigdl/nano/deps/openvino/pytorch/model.py

```diff
@@ -46,17 +46,23 @@ def __init__(self, model, input_sample=None, thread_num=None,
         :param **export_kwargs: will be passed to torch.onnx.export function.
         """
         ov_model_path = model
-        with TemporaryDirectory() as dir:
-            dir = Path(dir)
+        with TemporaryDirectory() as tmpdir:
+            tmpdir = Path(tmpdir)
             if isinstance(model, torch.nn.Module):
-                export(model, input_sample, str(dir / 'tmp.xml'), logging, **export_kwargs)
-                ov_model_path = dir / 'tmp.xml'
+                export(model, input_sample, str(tmpdir / 'tmp.xml'), logging, **export_kwargs)
+                ov_model_path = tmpdir / 'tmp.xml'
 
             self.ov_model = OpenVINOModel(ov_model_path, thread_num=thread_num, config=config)
             super().__init__(None)
         self.context_manager = generate_context_manager(accelerator="openvino",
                                                         precision="fp32",
                                                         thread_num=thread_num)
+        # patch original model's attr to current new model
+        if isinstance(model, torch.nn.Module):
+            for attr in dir(model):
+                if attr not in dir(self) and not attr.startswith('_') and not\
+                        isinstance(getattr(model, attr), torch.nn.Module):
+                    setattr(self, attr, getattr(model, attr))
```

Comment:
Refactor this to a utility method that everyone can call (instead of writing the same code multiple times).

Why do you skip all the attributes starts with '_'? A regular method may depend on these.

And we will save another copy of the original weight? (Maybe it's just a reference and does not matter?)

Reply:
will revise according to first two points.
and I don't quite understand the third point, another copy of original weight is ?

Reply:
> will revise according to first two points. and I don't quite understand the third point, another copy of original weight is ?

Both the weight of the new model (e.g., OV or ORT) and the original FP32 PyTorch model?

Reply:
1. for `Why do you skip all the attributes starts with '_'? A regular method may depend on these.`
After validation, actually we can just skip them, as these private attributes can be obtained implicitly through the original model, below is a case:
```
class A:
    def __init__(self):
        self._x = 1
        self.channels = 3
    
    def print(self):
        print(self)
        print(self._x + self.channels)


class B:
    def __init__(self, a):
        self.y = 1
        self.channels = a.channels
        self.print = a.print

a = A()
b = B(a)
a.print()
# <__main__.A object at 0x7f5766fa6c10>
# 4
b.print()
# <__main__.A object at 0x7f5766fa6c10>
# 4
```
2. about `save another copy of the original weight`
In fact, this PR don't support to get original attributes after save & load.
If we want to support this, it's not enough to just save model's weight, instead, we may need to save the whole model and then we can load attributes like `channels`. I feel this is a little complex, and the saved files will be larger. Do you think shall we support this ? @jason-dai @TheaperDeng